pkg/api/v1/types.go

```diff
@@ -2026,16 +2026,16 @@ type NodeSelector struct {
 // A null or empty node selector term matches no objects.
 type NodeSelectorTerm struct {
 	//Required. A list of node selector requirements. The requirements are ANDed.
-	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" protobuf:"bytes,1,rep,name=matchExpressions"`
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,rep,name=matchExpressions"`
```

Comment:
We need some way to change and make our API correct without cycling the version of everything that has a PodSpec in it.

One way of doing this is to have the client send the merge key and strategy that it actually used to calculate the patch as part of the patch itself.  The server would also send the recommended merge keys / strategies to the client as part of the open API before the client calculates the patch.

This would require supporting 2 versions of the strategy / key 
- "static / implicit" (will never change and even if it doesn't do the right thing)
- "dynamic / explicit" (explicitly communicated by the patch author and recommended by the server)

New clients:
- Fetch merge strategy / key obtained from open API
  - Already exists, but need to come up with way to make this deviate from the static tags
- Calculate patch request using this strategy / key
  - In progress, hopefully 1.9
- Include strategy / key in the patch request 
- Server applies patch using the provided strategy / key

Old clients:
- Use merge strategy / key obtained from tribal knowledge
- Calculate patch request using this strategy / key
- Don't include information about what the strategy / key is in the patch
- Server applies patch using implicit legacy key